src/main/java/io/cdap/plugin/snowflake/source/batch/SnowflakeMapToRecordTransformer.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.snowflake.source.batch;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.format.UnexpectedFormatException;
+import io.cdap.cdap.api.data.schema.Schema;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Transforms Snowflake row into {@link StructuredRecord}.
+ */
+public class SnowflakeMapToRecordTransformer {
+  private static final Logger LOG = LoggerFactory.getLogger(SnowflakeMapToRecordTransformer.class);
+
+  private final Schema schema;
+
+  public SnowflakeMapToRecordTransformer(Schema schema) {
+    this.schema = schema;
+  }
+
+  public StructuredRecord transform(Map<String, String> row) {
+    return getStructuredRecord(row, schema);
+  }
+
+  private StructuredRecord getStructuredRecord(Map<String, String> row, Schema schema) {
+    StructuredRecord.Builder builder = StructuredRecord.builder(schema);
+    row.entrySet().stream()
+      .filter(entry -> schema.getField(entry.getKey()) != null) // filter absent fields in the schema
+      .forEach(entry -> builder.set(
+        entry.getKey(),
+        convertValue(entry.getKey(), entry.getValue(), schema.getField(entry.getKey()).getSchema())));
+    return builder.build();
+  }
+
+  private Object convertValue(String fieldName, String value, Schema fieldSchema) {
```

Comment:
`convertValue` should be marked as nullable and make sure caller handles null values to avoid NPE

Reply:
Done